File: domain/src/main/java/com/mrp/sml/domain/repository/TransferHistoryRepository.java
package com.mrp.sml.domain.repository;

import com.mrp.sml.domain.model.TransferRecord;
import java.util.List;

public interface TransferHistoryRepository {
    List<TransferRecord> getTransferHistory();
    void saveTransferRecord(TransferRecord record);

    void observeTransferHistory(TransferHistoryListener listener);
    void removeTransferHistoryObserver(TransferHistoryListener listener);

    interface TransferHistoryListener {
        void onHistoryChanged(List<TransferRecord> history);
    }
}

File: domain/src/main/kotlin/com/mrp/sml/domain/repository/TransferHistoryRepository.kt
package com.mrp.sml.domain.repository

import com.mrp.sml.domain.model.TransferRecord
import kotlinx.coroutines.flow.Flow

interface TransferHistoryRepository {
    fun observeTransferHistory(): Flow<List<TransferRecord>>

    suspend fun saveTransferRecord(record: TransferRecord)
}
